experimental/adaptive-tool/README.md

```diff
@@ -1,57 +1,118 @@
-#### .lu documents
-- syntax highlighting
-![lu_synt	ax_highlighting](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/lu_syntax_highlighting.png)
-- diagnostic check
-![lu_diagnostic](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/lu_diagnostic.png)
-- completion 
-![lu_completion](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/lu_completion.gif)
+#### LU documents
+- Syntax highlighting
+![lu_syntax_highlighting](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/lu_syntax_highlighting.png)
+- Diagnostic check
+![lu_diagnostic](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/lu_diagnostic.png)
+- Completion 
+![lu_completion](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/lu_completion.gif)
+
+#### LG documents
+- Syntax highlighting
+![lg_syntax_highlighting](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/lg_syntax_highlighting.png)
+
+- Diagnostic check
+![lg_diagnostic](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/lg_diagnostic.gif)
+
+- Template reference hover
+![template_hover](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/template_hover.png)
+
+- Builtin function hover
+![function_hover](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/function_hover.png)
+
+- Builtin function and template suggestion
+![function_template_suggestion](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/function_template_suggestion.gif)
+
+- Structure property suggestion
+![structure_suggestion](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/structure_suggestion.gif)
 
-#### .lg documents
-- syntax highlighting
-![lg_syntax_highlighting](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/lg_syntax_highlighting.png)
+- Template definition
+![template_definition](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/template_definition.gif)
 
-- diagnostic check
-![lg_diagnostic](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/lg_diagnostic.gif)
+## Debugging Feature
+### Setting up and using Visual Studio Code to run client and server
+#### Setting up
```

Comment:
I think in here, we want to give an E2E setup of debugging .dialog for example, including, checking out which project (maybe testbotJson in dotnet), how to run, how to setup vscode like.